tests/python_package_test/test_dask.py

```diff
@@ -297,6 +299,10 @@ def test_classifier(output, task, boosting_type, tree_learner, cluster):
             assert_eq(p1_local, p2)
             assert_eq(p1_local, y)
 
+        # extra predict() parameters should be passed through correctly
+        with pytest.raises(AssertionError):
+            assert_eq(p1_raw, p1_first_iter)
```

Comment:
on `master`, this test and the others added in this PR pass for all Dask Dataframe inputs and fail for all Dask Array inputs

output of `cd tests/python_package_test/ && pytest test_dask.py` on `master`:

```text
====================================== short test summary info ======================================
FAILED test_dask.py::test_classifier[data-gbdt-binary-classification-array] - Failed: DID NOT RAIS...
FAILED test_dask.py::test_classifier[data-gbdt-binary-classification-scipy_csr_matrix] - Failed: D...
FAILED test_dask.py::test_classifier[data-gbdt-multiclass-classification-array] - Failed: DID NOT ...
FAILED test_dask.py::test_classifier[data-gbdt-multiclass-classification-scipy_csr_matrix] - Faile...
FAILED test_dask.py::test_classifier[data-dart-binary-classification-array] - Failed: DID NOT RAIS...
FAILED test_dask.py::test_classifier[data-dart-binary-classification-scipy_csr_matrix] - Failed: D...
FAILED test_dask.py::test_classifier[data-dart-multiclass-classification-array] - Failed: DID NOT ...
FAILED test_dask.py::test_classifier[data-dart-multiclass-classification-scipy_csr_matrix] - Faile...
FAILED test_dask.py::test_classifier[data-goss-binary-classification-array] - Failed: DID NOT RAIS...
FAILED test_dask.py::test_classifier[data-goss-binary-classification-scipy_csr_matrix] - Failed: D...
FAILED test_dask.py::test_classifier[data-goss-multiclass-classification-array] - Failed: DID NOT ...
FAILED test_dask.py::test_classifier[data-goss-multiclass-classification-scipy_csr_matrix] - Faile...
FAILED test_dask.py::test_classifier[data-rf-binary-classification-array] - Failed: DID NOT RAISE ...
FAILED test_dask.py::test_classifier[data-rf-binary-classification-scipy_csr_matrix] - Failed: DID...
FAILED test_dask.py::test_classifier[data-rf-multiclass-classification-array] - Failed: DID NOT RA...
FAILED test_dask.py::test_classifier[data-rf-multiclass-classification-scipy_csr_matrix] - Failed:...
FAILED test_dask.py::test_classifier[voting-gbdt-binary-classification-array] - Failed: DID NOT RA...
FAILED test_dask.py::test_classifier[voting-gbdt-binary-classification-scipy_csr_matrix] - Failed:...
FAILED test_dask.py::test_classifier[voting-gbdt-multiclass-classification-array] - Failed: DID NO...
FAILED test_dask.py::test_classifier[voting-gbdt-multiclass-classification-scipy_csr_matrix] - Fai...
FAILED test_dask.py::test_classifier[voting-dart-binary-classification-array] - Failed: DID NOT RA...
FAILED test_dask.py::test_classifier[voting-dart-binary-classification-scipy_csr_matrix] - Failed:...
FAILED test_dask.py::test_classifier[voting-dart-multiclass-classification-array] - Failed: DID NO...
FAILED test_dask.py::test_classifier[voting-dart-multiclass-classification-scipy_csr_matrix] - Fai...
FAILED test_dask.py::test_classifier[voting-goss-binary-classification-array] - Failed: DID NOT RA...
FAILED test_dask.py::test_classifier[voting-goss-binary-classification-scipy_csr_matrix] - Failed:...
FAILED test_dask.py::test_classifier[voting-goss-multiclass-classification-array] - Failed: DID NO...
FAILED test_dask.py::test_classifier[voting-goss-multiclass-classification-scipy_csr_matrix] - Fai...
FAILED test_dask.py::test_classifier[voting-rf-binary-classification-array] - Failed: DID NOT RAIS...
FAILED test_dask.py::test_classifier[voting-rf-binary-classification-scipy_csr_matrix] - Failed: D...
FAILED test_dask.py::test_classifier[voting-rf-multiclass-classification-array] - Failed: DID NOT ...
FAILED test_dask.py::test_classifier[voting-rf-multiclass-classification-scipy_csr_matrix] - Faile...
FAILED test_dask.py::test_classifier_pred_contrib[multiclass-classification-scipy_csr_matrix] - At...
FAILED test_dask.py::test_regressor[data-gbdt-array] - Failed: DID NOT RAISE <class 'AssertionError'>
FAILED test_dask.py::test_regressor[data-gbdt-scipy_csr_matrix] - Failed: DID NOT RAISE <class 'As...
FAILED test_dask.py::test_regressor[data-dart-array] - Failed: DID NOT RAISE <class 'AssertionError'>
FAILED test_dask.py::test_regressor[data-dart-scipy_csr_matrix] - Failed: DID NOT RAISE <class 'As...
FAILED test_dask.py::test_regressor[data-goss-array] - Failed: DID NOT RAISE <class 'AssertionError'>
FAILED test_dask.py::test_regressor[data-goss-scipy_csr_matrix] - Failed: DID NOT RAISE <class 'As...
FAILED test_dask.py::test_regressor[data-rf-array] - Failed: DID NOT RAISE <class 'AssertionError'>
FAILED test_dask.py::test_regressor[data-rf-scipy_csr_matrix] - Failed: DID NOT RAISE <class 'Asse...
FAILED test_dask.py::test_regressor[voting-gbdt-array] - Failed: DID NOT RAISE <class 'AssertionEr...
FAILED test_dask.py::test_regressor[voting-gbdt-scipy_csr_matrix] - Failed: DID NOT RAISE <class '...
FAILED test_dask.py::test_regressor[voting-dart-array] - Failed: DID NOT RAISE <class 'AssertionEr...
FAILED test_dask.py::test_regressor[voting-dart-scipy_csr_matrix] - Failed: DID NOT RAISE <class '...
FAILED test_dask.py::test_regressor[voting-goss-array] - Failed: DID NOT RAISE <class 'AssertionEr...
FAILED test_dask.py::test_regressor[voting-goss-scipy_csr_matrix] - Failed: DID NOT RAISE <class '...
FAILED test_dask.py::test_regressor[voting-rf-array] - Failed: DID NOT RAISE <class 'AssertionError'>
FAILED test_dask.py::test_regressor[voting-rf-scipy_csr_matrix] - Failed: DID NOT RAISE <class 'As...
FAILED test_dask.py::test_ranker[data-gbdt-None-array] - Failed: DID NOT RAISE <class 'AssertionEr...
FAILED test_dask.py::test_ranker[data-gbdt-group1-array] - Failed: DID NOT RAISE <class 'Assertion...
FAILED test_dask.py::test_ranker[data-dart-None-array] - Failed: DID NOT RAISE <class 'AssertionEr...
FAILED test_dask.py::test_ranker[data-dart-group1-array] - Failed: DID NOT RAISE <class 'Assertion...
FAILED test_dask.py::test_ranker[data-goss-None-array] - Failed: DID NOT RAISE <class 'AssertionEr...
FAILED test_dask.py::test_ranker[data-goss-group1-array] - Failed: DID NOT RAISE <class 'Assertion...
FAILED test_dask.py::test_ranker[data-rf-None-array] - Failed: DID NOT RAISE <class 'AssertionError'>
FAILED test_dask.py::test_ranker[data-rf-group1-array] - Failed: DID NOT RAISE <class 'AssertionEr...
FAILED test_dask.py::test_ranker[voting-gbdt-None-array] - Failed: DID NOT RAISE <class 'Assertion...
FAILED test_dask.py::test_ranker[voting-gbdt-group1-array] - Failed: DID NOT RAISE <class 'Asserti...
FAILED test_dask.py::test_ranker[voting-dart-None-array] - Failed: DID NOT RAISE <class 'Assertion...
FAILED test_dask.py::test_ranker[voting-dart-group1-array] - Failed: DID NOT RAISE <class 'Asserti...
FAILED test_dask.py::test_ranker[voting-goss-None-array] - Failed: DID NOT RAISE <class 'Assertion...
FAILED test_dask.py::test_ranker[voting-goss-group1-array] - Failed: DID NOT RAISE <class 'Asserti...
FAILED test_dask.py::test_ranker[voting-rf-None-array] - Failed: DID NOT RAISE <class 'AssertionEr...
FAILED test_dask.py::test_ranker[voting-rf-group1-array] - Failed: DID NOT RAISE <class 'Assertion...
================ 65 failed, 214 passed, 3 skipped, 260 warnings in 120.86s (0:02:00) ================
```

Reply:
```suggestion
            assert_eq(p1_raw, p1_first_iter_raw)
```